Fix: store autosave settings under the new names

TrueBrain opened the pull request:
## Motivation / Problem

While looking for a way to allow aliases for `OMANY`, I noticed autosave was storing the wrong settings values. And that we have a way to allow aliases.

## Description

Instead of storing the old `monthly`, `yearly`, etc, store it in minutes.

This keeps the original setting in tact, and renames the new one. This allows switching back between old and new versions without annoying warnings.

Based on comments and feedback, this PR grew a bit bigger to supply a framework for future conversions too. Also makes it a bit more readable :)

## Limitations

The GUI doesn't allow for anything between the predefined settings. This means if the user changes the value in the config to something other than 0/10/30/60/120, it will work, till the Game Options are opened again.


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

PeterN commented on src/settings.cpp:
```diff
@@ -1262,6 +1264,27 @@ void LoadFromConfig(bool startup)
+					}
+				}
+			}
+		}
```
Should we check if the new setting already exists?

Not sure, but I think if you load a new (and change the value) then old then new version it will reset your `autosave_interval` to whatever `autosave` maps to?

TrueBrain replied:
Yes and no. It depends :D

Configuration-files have a version, and this check ofc validates if you are loading a configuration with an older version. So what can happen is:

- You have a v4 config.
- You load a client with v5 support. It converts the `autosave` from v4 to v5.
- You now load a client with v4 support. It will resave the config as a v4.
- Now if you load a client with v5 support again, it will convert the `autosave` again, as that is what the version asks.

I am not sure this is a bad thing. How ever you slice it, if people use the same configuration with multiple versions, you will always have these weird edge-cases to resolve. This is the simplest way to approach it. What it means that if someone changes the autosave in the v4 config, it will always have an effect on v5.

If you would be to check if `autosave_interval` already exists, it won't convert it again. This can be good .. this can be bad.

So far, for all the other settings, we opted for not checking if the new setting exists. The setting in the older config wins from the newer.

PeterN replied:
Problem with this is it will override the setting in the new version even if you didn't change the setting in the old version.

And I do often load an old version just to check behaviour of something else.

So I'm not sure I like the current way...

TrueBrain replied:
Takes a bit more code, so I generalized the approach a bit. There was one other setting that did the same, so converted that. That makes this PR a bit bigger than the exact context of the PR, but helps future-us in not having this talk again to figure out what we actually want :)

TrueBrain committed:
Codechange: use BoolSettingDesc to parse settings in conversion code

TrueBrain committed:
Codechange: refactor setting conversions and make sure it only happens once

By forcing it only happens once, the first time you upgrade to
a newer client, means you are free to jump between older and newer
versions after that. As they will not resync, the newer setting
can take on any of the (newer) values, without breaking the old
client. And when going to an old client and back, it doesn't
change it back to the converted value anymore.

TrueBrain committed:
Fix: store autosave settings under the new names
